src/runtime/classobj.cpp

```diff
@@ -506,6 +515,76 @@ Box* instanceDelitem(Box* _inst, Box* key) {
     return runtimeCall(delitem_func, ArgPassSpec(1), key, NULL, NULL, NULL, NULL);
 }
 
+Box* instanceGetslice(Box* _inst, Box* i, Box* j) {
+    RELEASE_ASSERT(_inst->cls == instance_cls, "");
+    BoxedInstance* inst = static_cast<BoxedInstance*>(_inst);
+
+    static BoxedString* getslice_str = internStringImmortal("__getslice__");
+    Box* getslice_func = NULL;
+
+    try {
+        getslice_func = _instanceGetattribute(inst, getslice_str, false);
+    } catch (ExcInfo e) {
+        if (!e.matches(AttributeError))
+            throw e;
+    }
+
+    if (getslice_func == NULL) {
+        Box* slice = static_cast<Box*>(createSlice(i, j, None));
+        return instanceGetitem(inst, slice);
+    }
+
+    return runtimeCall(getslice_func, ArgPassSpec(2), i, j, NULL, NULL, NULL);
+}
```

Comment:
Minor (not worth changing), but we could have directly copied CPython's instance_slice.  It looks like the conversion got rid of some overhead, but I've been liking the approach of starting out with a 100% copy and then optimize as needed.